### ISSUE TITLE
New quest: Smoking?

### ISSUE THREAD

**Issue by rugk** [streetcomplete/StreetComplete#539]
For cafes/restaurants/etc the [smoking tag](https://wiki.openstreetmap.org/wiki/Key:smoking) is useful.

Of course, this needs some country-specific research as in Germany, e.g. restaurants are not allowed to have smokers and non-smokers rooms together, but they *must be* split. (So smoking=yes is not really possible.)

**Comment by Binnette** [comment]
In france smoking is not allowed in public areas since 2008. But some restaurant/bar have an 'isolated' room in wich smoking is allowed.

**Comment by westnordost** [comment]
In Germany there are places which explicitly allow smoking. (And if it is not advertised, it usually means smoking=no)

**Comment by alejandroscf** [comment]
In Spain is forbidden to smoke in all restaurants, bars and pubs, also in public/government buildings.

**Comment by westnordost** [comment]
Since when? When I lived there (2011), smoking was forbidden "by default" and every place that allowed smoking had to explicitly have some kind of note/certificate/sign (I don't remember) posted somewhere visible.

**Comment by majkaz** [comment]
The same Czech republic - no smoking in restaurants and pubs at all (new in 2017).

**Comment by pietervdvn** [comment]
In Belgium, smoking is forbidden as well. It is only allowed in a separated room in some restaurants.

**Comment by rory** [comment]
Smoking has been banned in all enclosed workplaces in Ireland since ~2003. So this would always be `smoking=no` in Ireland.

**Comment by voschix** [comment]
In Italy, all indoors public places are no.smoking by law. Very few places have dedicated smoker rooms.

**Comment by hmartink** [comment]
I would support adding such a quest. I'm not sure general rules (like "smoking=yes" not possible in Germany) are sensible/possible in SC. If it is possible, then of course this quest might be superfluous in some areas (if really all options, like smoking in front of the building) are prohibited.

And if a mapper does not like being asked (e.g. if the law does not allow several options), then she could simply use the settings to avoid being asked this quest.
As the comments show, there are many general rules which usually allow some kind of configurability (e.g. separate rooms, smoking outside). Beeing a non-smoker, I would highly appreciate obtaining the information beforehand. But if I were a smoke, this information would be also useful (e.g. if in Italy or Belgium I would look for those rare places with dedicated smoker rooms). 

This tag is also not difficult to obtain: At least the places I experienced (and I travel quite a bit) the smoking regulation is usually posted near the entrance door, either outside or (in restaurants) inside. So when mapping the opening times I can often easily map this information as well, and if I map the menu or the accessibility of rest rooms (where I would enter the building) I could map the rest. 

Finally, when I used the web frontend of OSM (which I rarely do) I noticed that I could enter only a limited set of choices for possible answers. So it should be easily implementable in SC, the only downside is that most likely the choice needs to contain the different options as text, as I'm not sure you could put in good symbols for each choice. But otherwise, quite similar to e.g. the road paving selection.

So in conclusion: The rules appear to be general but with (possible) exceptions in various countries,  (usefulness), the mapping is usually not complicated (saving mappers time) and [at least for a non programmer] the implementation does not appear to be very complicated.

**Comment by goldfndr** [comment]
https://en.wikipedia.org/wiki/List_of_smoking_bans has a comprehensive list of references.

**Comment by westnordost** [comment]
Thanks for the list, this makes it much easier to decide for which countries to show this quest!

It should only be asked for pubs, bars and maybe restaurants and maybe fast food restaurants.

**Comment by mnalis** [comment]
Smoking quest  would be really useful, and not hard to implement. If problematic, disable it by default (like vegan etc). Just please don't country-block it.

Because law is sometimes imprecise, wikipedia is not updated all the time, the laws changes often and has exceptions and transition periods, and mostly the status on the ground is what should really be mapped -- as I customer allergic to smoke, it matters to me not one iota if there is backdoor in the law that allows smoking even when it is "prohibited by default". For example, the wikipedia claim of ["16 establishments in whole of Croatia with allowed smoking"](https://en.wikipedia.org/wiki/List_of_smoking_bans#Croatia) is both obsolete, misleading, and laughable to anyone living here -- in 2020, there are more than 16 smoking pubs/cafes in my neighborhood alone, let alone whole of the country!

**Comment by matkoniecz** [comment]
> Because law is sometimes imprecise, wikipedia is not updated all the time, the laws changes often and has exceptions

As soon as someone reports that country block is not valid - we can change metadata.

Though, if it would be disabled by default - it can be very simple to implement it, like kosher quest.

**Comment by mnalis** [comment]
Sooooo, if it really is that simple that my copy/paste skill would mostly suffice, I'd like to try my hand in making a basic PR for this "smoking" quest? (unless someone else really wants to do it, of course)

**Comment by matkoniecz** [comment]
I created https://docs.google.com/spreadsheets/d/1RpLbXVP0PKDLUV6w1pyCSlyJ4tjoTF5jJHaAtfAGJBk/edit?usp=sharing (editing to anyone with link, hopefully it will work and not will be overrun by spam edits)

Processing https://en.wikipedia.org/wiki/List_of_smoking_bans may be enough to fill it

Looking it at it again it seems clear that it would be one more disabled by default quest (I need to look at statistics how many people enable them and consider is it a good idea to make enabling them easier).

@mnalis 

Not sure are you still interested but it should be fairly easy. Main effort is researching legal situation and disabling it in countries where it is pointless (due to 100% ban on smoking or distinct lack of bans anywhere or something)

**Comment by mnalis** [comment]
I've been looking into it, reading up on [CONTRIBUTING_A_NEW_QUEST.md](https://github.com/matkoniecz/Zazolc/blob/so-you-want-to-make-a-new-quest/CONTRIBUTING_A_NEW_QUEST.md) ;  but am unsure which dialog type would be best.

- I could go with simple `yes`/`no` like the `lit quest` and call it a day

However, [smoking=*](https://wiki.openstreetmap.org/wiki/Key:smoking) has several other values besides `yes`/`no` - with especially `smoking=outside` seemingly [very popular](https://taginfo.openstreetmap.org/keys/smoking#values), but there are also values `dedicated` / `separated` / `isolated`.

As it seems very hard to imagine pictures which would depict various values (especially if we want to anonymize people faces), I don't think that approach would work well. 

- So I'm currently thinking of doing 3-most-popular-value text answer (like in `Wheelchair quest` = `yes`/`no`/`partially`); so something like:

```
Is smoking allowed here?
- yes
- no
- only outside
```

As `dedicated` is likely already marked as such (as it is not regular cafe etc but more of a club dedicated to smoking) I don't think it is a big problem to omit it; and while `separated` / `isolated` are adding more detail than `yes`, it does not seem crucial, and it is hard to explain the difference if SC quest format - there is always `Leave a note` as a catch-all.

Any comments / suggestions on that?

**Comment by voschix** [comment]
Annotations:

   - What about outdoor no-smoking like train platforms in Germany, and
   larger no smoking areas in  California
   - No--smoking rules in the USA vary from State to State (
   https://en.wikipedia.org/wiki/List_of_smoking_bans_in_the_United_States)


<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Virus-free.
www.avast.com
<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

**Comment by westnordost** [comment]
How can "outside" not be allowed? Is it defined differently than I think?

**Comment by mnalis** [comment]
- I would (at least initially) only ask on `amenity=cafe`, `amenity=pub`, `amenity=bar`, `amenity=restaurant`, `amenity=nightclub`, `amenity=biergarten` and maybe `leisure=outdoor_seating` as they may differ in countries where smoking is (at least partially) allowed. 
  I think that if smoking it is disallowed at hospitals, train stations and similar, than it is disallowed on all of them in that region/country, so no need to spam user with questions about it.
- Additionally, I'd likely make the quest disabled by default, for the reason @voschix mentions too - it is very hard to find correctly exactly where it is useful and where not, so leaving that to user decision seems best (not to mention easiest :sweat_smile:)

> How can "outside" not be allowed? Is it defined differently than I think?

@westnordost Well, I'd have to know what you think to answer that. According to wiki `smoking=outside` is defined as _**"Smoking is allowed on the property but only in outside areas"**_. Respectively, it may be forbidden on both parts of the property (both inside and outside). 

So for there is a cafe which has outdoor seating in their ownership, and they can decide to limit what is allowed at that outside seating too. It definitely does not refer to _"whole rest of the outside world except this cafe"_ or anything like that. So they cannot control if you stand 5m outside of their fence/ and smoke a cigar, but they can forbid you to do it inside their garden/outside seating. (and can replace fence/free area with hedge/wall if they find such nearby smoking or other situation like traffic problematic enough) 

I don't know about Germany but in Croatia that happens relatively often - some pub/cafe/restaurant wants to cater to non-smokers specifically (either because of their beliefs, or purely by economic reasons for example if there is plenty of competing smoking cafes around,  or some behaviour might even be required by law eg. you can be registered as healthy refreshments provider near the gym or so, but then you're forbidden from allowing smoking or selling alcohol, on all of that property, regardless if it is inside or outside)

**Comment by matkoniecz** [comment]
> Is smoking allowed here?
> - yes
> - no
> - only outside

It seems to work, though I did only quite limited research

> I've been looking into it, reading up on [CONTRIBUTING_A_NEW_QUEST.md](https://github.com/matkoniecz/Zazolc/blob/so-you-want-to-make-a-new-quest/CONTRIBUTING_A_NEW_QUEST.md)

Please let me know in its pull request if anything is unclear about anything!
